drivers/sensor/memsic/mmc56x3/mmc56x3.c

```diff
@@ -0,0 +1,478 @@
+/*
+ * Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+#include <math.h>
+
+#include "mmc56x3.h"
+
+LOG_MODULE_REGISTER(MMC56X3, CONFIG_SENSOR_LOG_LEVEL);
+
+K_TIMER_DEFINE(meas_req_timer, NULL, NULL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "MMC56X3 driver enabled without any devices"
+#endif
+
+static inline int mmc56x3_bus_check(const struct device *dev)
+{
+	const struct mmc56x3_dev_config *config = dev->config;
+
+	return config->bus_io->check(&config->bus);
+}
+
+static inline int mmc56x3_reg_read(const struct device *dev, uint8_t reg, uint8_t *buf, int size)
+{
+	const struct mmc56x3_dev_config *config = dev->config;
+
+	return config->bus_io->read(&config->bus, reg, buf, size);
+}
+
+static inline int mmc56x3_reg_write(const struct device *dev, uint8_t reg, uint8_t val)
+{
+	const struct mmc56x3_dev_config *config = dev->config;
+
+	return config->bus_io->write(&config->bus, reg, val);
+}
+
+static inline int mmc56x3_raw_read(const struct device *dev, uint8_t *buf, int size)
+{
+	const struct mmc56x3_dev_config *config = dev->config;
+
+	return config->bus_io->raw_read(&config->bus, buf, size);
+}
+
+static inline int mmc56x3_raw_write(const struct device *dev, uint8_t *buf, int size)
+{
+	const struct mmc56x3_dev_config *config = dev->config;
+
+	return config->bus_io->raw_write(&config->bus, buf, size);
+}
+
+static int mmc56x3_chip_set_auto_self_reset(const struct device *dev, bool auto_sr)
+{
+	struct mmc56x3_data *data = dev->data;
+	struct mmc56x3_config *config = &data->config;
+
+	if (auto_sr) {
+		data->ctrl0_cache |= MMC56X3_CMD_AUTO_SELF_RESET_EN;
+	} else {
+		data->ctrl0_cache &= ~MMC56X3_CMD_AUTO_SELF_RESET_EN;
+	}
+
+	int ret = mmc56x3_reg_write(dev, MMC56X3_REG_INTERNAL_CTRL_0, data->ctrl0_cache);
+
+	if (ret < 0) {
+		LOG_DBG("Setting auto_sr_en bit failed: %d", ret);
+	} else {
+		config->auto_sr = auto_sr;
+	}
+
+	return ret;
+}
+
+/* Set ODR to 0 to turn off */
+static int mmc56x3_chip_set_continuous_mode(const struct device *dev, uint16_t odr)
+{
+	struct mmc56x3_data *data = dev->data;
+	struct mmc56x3_config *config = &data->config;
+
+	if (odr > 255) {
+		odr = 1000;
+		data->ctrl2_cache |= MMC56X3_CMD_HPOWER;
+	}
+
+	int ret = mmc56x3_reg_write(dev, MMC56X3_REG_INTERNAL_ODR, odr);
```

Comment:
Not sure you want to write 1000 in a uint8_t register?

Reply:
Dumb mistake, great catch. Fixed